### PR TITLE
RN 0.67.5 + Android SDK 31

### PR DESCRIPTION
**Pull request by stackchain**
- Google play store target SDK now is 31, min 23
- Apple store min iOS is 11
---

### New requirements

- Install `jdk11` (replacement for `jdk8`)
- Remove `node_modules` (broken `yarn.lock`) + `yarn cache clean && yarn install`
- Update pods `cd ios && pod update`
- Install `NDK 24.*`

### CODE REVIEW

**Review by wolverineks — CHANGES_REQUESTED** [review]
https://react-native-community.github.io/upgrade-helper/?from=0.64.4&to=0.67.5

**Comment by stackchain** [comment]
> https://react-native-community.github.io/upgrade-helper/?from=0.64.4&to=0.67.5

yap.

**Comment by wolverineks** [comment]
this pr is impossibly big
i dont see a way to review this
can you break it into 10+ different prs?

File: .eslintrc.js [hunk 1]
```diff
@@ -1,73 +1,27 @@
...
-          'error',
-          {allowString: false, allowNumber: false, allowNullableBoolean: true},
-        ],
+        '@typescript-eslint/no-shadow': ['error'],
+        'no-shadow': 'off',
+        'no-undef': 'off',
       },
     },
   ],
-  extends: [
```
Comment (stackchain) [inline .eslintrc.js:19]: check against all comminity rules

File: __mocks__/@emurgo/react-native-haskell-shelley.js [hunk 2]
```diff
@@ -1,2 +1 @@
-// @flow
```
Comment (stackchain) [inline __mocks__/@emurgo/react-native-haskell-shelley.js:1]: clean up

File: __mocks__/@react-native-async-storage/async-storage.js [hunk 3]
```diff
@@ -1 +1 @@
-export default from '@react-native-async-storage/async-storage/jest/async-storage-mock'
```
Comment (stackchain) [inline __mocks__/@react-native-async-storage/async-storage.js:1]: bugfix

File: android/app/build.gradle [hunk 4]
```diff
@@ -200,36 +209,36 @@ android {
...
         }
         nightly {
             dimension "version"
             applicationIdSuffix ".nightly"
             applicationId 'com.emurgo'
-            minSdkVersion 21
-            compileSdkVersion = 30
-            targetSdkVersion 30
+            minSdkVersion 23
+            compileSdkVersion = 31
```
Comment (stackchain) [inline android/app/build.gradle:231]: check which version will take over during the compiling

File: android/app/build.gradle [hunk 5]
```diff
@@ -150,7 +154,7 @@ android {
         applicationId "com.emurgo"
         minSdkVersion rootProject.ext.minSdkVersion
         targetSdkVersion rootProject.ext.targetSdkVersion
-        versionCode 356
+        versionCode 358
```
Comment (stackchain) [inline android/app/build.gradle:157]: remove compile options

File: android/app/src/main/java/com/emurgo/MainApplication.java [hunk 6]
```diff
@@ -4,6 +4,7 @@
 import android.content.Context;
 import com.facebook.react.PackageList;
 import com.facebook.react.ReactApplication;
+import com.igorbelyayev.rnlocalresource.RNLocalResourcePackage;
```
Comment (stackchain) [inline android/app/src/main/java/com/emurgo/MainApplication.java:7]: check if we can get rid off

File: android/build.gradle [hunk 7]
```diff
@@ -2,13 +2,13 @@
 
 buildscript {
     ext {
-        buildToolsVersion = "30.0.2"
-        minSdkVersion = 21
-        compileSdkVersion = 30
-        targetSdkVersion = 30
+        buildToolsVersion = "30.0.3"
+        minSdkVersion = 23
+        compileSdkVersion = 31
```
Comment (stackchain) [inline android/build.gradle:7]: update to 30

File: babel.config.js [hunk 8]
```diff
@@ -1,5 +1,10 @@
 module.exports = {
   presets: ['module:metro-react-native-babel-preset'],
+  env: {
```
Comment (stackchain) [inline babel.config.js:3]: package update

File: e2e/utils.js [hunk 9]
```diff
@@ -5,7 +5,7 @@
  * hack to get text from element (feature not available in detox)
  *  see issue https://github.com/wix/detox/issues/445
  */
-export async function readTextValue(testID: string) {
```
Comment (stackchain) [inline e2e/utils.js:8]: clean up

File: package.json [hunk 10]
```diff
@@ -77,15 +77,15 @@
     ]
   },
   "resolutions": {
-    "react-devtools-core": "4.14.0",
```
Comment (stackchain) [inline package.json:80]: version of devtools -> is working

File: src/Catalyst/ConfirmVotingTx.tsx [hunk 11]
```diff
@@ -61,7 +61,7 @@ export const ConfirmVotingTx = ({
           value={formatTokenWithSymbol(new BigNumber(Amounts.getAmount(votingRegTx.fee, '').quantity), tokenInfo)}
           label={strings.fees}
           editable={false}
-          autoComplete={false}
+          autoComplete="off"
```
Comment (stackchain) [inline src/Catalyst/ConfirmVotingTx.tsx:64]: react-native stuff

File: src/Dashboard/Dashboard.tsx [hunk 12]
```diff
@@ -1,4 +1,3 @@
-/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (stackchain) [inline src/Dashboard/Dashboard.tsx:1]: clean up

File: src/Dashboard/DashboardNavigator.tsx [hunk 13]
```diff
@@ -32,7 +31,7 @@ export const DashboardNavigator = () => {
         component={Dashboard}
         options={{
           title: walletName,
-          headerRight: () => <HeaderRight />,
+          headerRight: HeaderRight,
```
Comment (stackchain) [inline src/Dashboard/DashboardNavigator.tsx:34]: lint rule

File: src/HW/LedgerConnect/DeviceItem/DeviceItem.stories.tsx [hunk 14]
```diff
@@ -12,10 +12,10 @@ const device: Device = {
 
 const doTheThing = () => new Promise((resolve) => setTimeout(resolve, 1000))
 
-const onSelect = async (device) => {
-  action('onSelect start')(device)
+const onSelect = async (selectedDevice) => {
```
Comment (stackchain) [inline src/HW/LedgerConnect/DeviceItem/DeviceItem.stories.tsx:15]: no shadow - var

File: tsconfig.json [hunk 15]
```diff
@@ -1,30 +1,66 @@
 {
   "compilerOptions": {
```
Comment (stackchain) [inline tsconfig.json:2]: please check

File: src/legacy/storage.ts [hunk 16]
```diff
@@ -54,7 +54,7 @@ export const remove = async (path: string) => {
   try {
     await AsyncStorage.removeItem(path)
   } catch (error) {
-    console.warn(`Missing storage key ${path}`)
+    Logger.warn(`Missing storage key ${path}`)
```
Comment (stackchain) [inline src/legacy/storage.ts:57]: ling rule - no console

**Comment by stackchain** [comment]
#2238

**Pull request closed without merging**